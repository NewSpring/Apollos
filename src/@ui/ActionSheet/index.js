export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
