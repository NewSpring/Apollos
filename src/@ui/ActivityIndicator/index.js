import { ActivityIndicator as NativeActivityIndicator, StyleSheet } from 'react-native';
import styled from '@ui/styled';

const ActivityIndicator = styled({
  flex: 1,
  ...StyleSheet.absoluteFillObject,
  alignSelf: 'center',
}, 'ActivityIndicator')(NativeActivityIndicator);

export {
  ActivityIndicator as default,
  NativeActivityIndicator as InlineActivityIndicator,
};
