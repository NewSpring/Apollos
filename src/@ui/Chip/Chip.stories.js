import React from 'react';

import { storiesOf } from '@storybook/react-native';
import FlexedView from '@ui/FlexedView';

import Chip from './';

storiesOf('Chip', module)
  .add('default', () => (
    <FlexedView style={{ flexWrap: 'wrap', flexDirection: 'row' }}>
      <Chip
        onPress={() => {}}
        title="I'm just a poor chip"
      />
      <Chip
        onPress={() => {}}
        icon="close"
        title="I need no sympathy"
      />
      <Chip
        onPress={() => {}}
        selected
        title="Easy come"
      />
      <Chip
        onPress={() => {}}
        icon="close"
        selected
        title="Easy go"
      />
      <Chip
        onPress={() => {}}
        icon="arrow-up"
        title="Litte high"
      />
      <Chip
        onPress={() => {}}
        type="secondary"
        icon="arrow-down"
        title="Litte low"
      />
      <Chip
        title="📍💨?"
      />
      <Chip
        selected
        title="¯\_(ツ)_/¯"
      />
    </FlexedView>
  ));

