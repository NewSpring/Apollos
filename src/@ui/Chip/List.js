import { View } from 'react-native';
import styled from '@ui/styled';

const ChipList = styled({
  flexDirection: 'row',
  alignItems: 'center',
  flexWrap: 'wrap',
}, 'Chip.List')(View);

export default ChipList;
