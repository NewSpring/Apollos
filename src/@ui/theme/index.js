export { default as ThemeProvider } from './ThemeProvider';
export { default as withTheme } from './withTheme';
export { ThemeMixin, withThemeMixin } from './mixins';
