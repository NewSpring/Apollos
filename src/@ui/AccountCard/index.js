import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { Platform } from 'react-native';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5 } from '@ui/typography';
import CategoryLabel from '@ui/CategoryLabel';
import Icon from '@ui/Icon';
import last4 from '@utils/last4';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    accountNumber: PropTypes.number,
    accountType: PropTypes.number,
    iconSize: PropTypes.number,
  }),
  defaultProps({
    isLoading: false,
    accountNumber: '',
    accountType: 'creditCard',
    iconSize: undefined,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = styled({
  flex: 1.66,
})(CardContent);

const RightColumn = styled(({ theme }) => ({
  paddingRight: theme.sizing.baseUnit,
  alignItems: 'flex-end',
  justifyContent: 'center',
  ...Platform.select({
    web: {
      position: 'relative',
    },
  }),
}))(FlexedView);

const AccountCard = enhance(({
  title,
  isLoading,
  accountNumber,
  accountType,
  iconSize,
  ...otherProps
}) => (
  <Card isLoading={isLoading} {...otherProps}>
    <HorizontalLayout>
      <LeftColumn>
        <H5>{startCase(toLower(title))}</H5>
        <CategoryLabel
          label={last4(accountNumber)}
          icon={accountType === 'bankAccount' ? 'bank' : 'credit'}
          isLoading={isLoading}
        />
      </LeftColumn>

      <RightColumn>
        <Icon name="ArrowNext" size={iconSize} />
      </RightColumn>
    </HorizontalLayout>
  </Card>
));

export default AccountCard;
