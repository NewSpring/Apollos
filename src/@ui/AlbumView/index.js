import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import { View, Platform, StyleSheet } from 'react-native';
import PaddedView from '@ui/PaddedView';
import ProgressiveImage from '@ui/ProgressiveImage';
import { H5, H7 } from '@ui/typography';
import { withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';

const Container = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.default,
}), 'AlbumView.Container')(View);

const InfoWrapper = styled({
  flexDirection: 'row', alignItems: 'stretch',
}, 'AlbumView.InfoWrapper')(PaddedView);

const AlbumArt = styled({
  position: 'relative',
  width: '33%',
  aspectRatio: 1,
  ...Platform.select({
    web: {
      height: 0,
      paddingTop: '100%',
    },
  }),
}, 'AlbumView.AlbumArt')(ProgressiveImage);

const BlurredImage = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  opacity: theme.alpha.low,
}), 'AlbumView.BlurredImage')(ProgressiveImage);

const TitleWrapper = styled({
  width: '66%',
  justifyContent: 'space-around',
}, 'AlbumView.TitleWrapper')(PaddedView);

const Title = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Title')(H5);

const Artist = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Artist')(H7);

const enhance = compose(
  withThemeMixin({ type: 'dark' }),
  pure,
  setPropTypes({
    title: PropTypes.string,
    artist: PropTypes.string,
    blurredImage: ProgressiveImage.propTypes.source,
    albumImage: ProgressiveImage.propTypes.source,
    isLoading: PropTypes.bool,
  }),
);

const AlbumView = enhance(({
  title, artist, blurredImage, albumImage, isLoading = false,
}) => (
  <Container>
    {(blurredImage) ? <BlurredImage source={blurredImage} /> : null}
    <InfoWrapper>
      <AlbumArt isLoading={isLoading} source={albumImage} />
      <TitleWrapper>
        <Title>{title}</Title>
        <Artist>{artist}</Artist>
      </TitleWrapper>
    </InfoWrapper>
  </Container>
));

export default AlbumView;
