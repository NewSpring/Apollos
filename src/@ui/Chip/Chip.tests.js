import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@ui/theme';

import Chip from './';

describe('The Chip component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Chip title="My Button!" onPress={() => {}} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render an icon', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Chip icon="like" title="Heart!!!" />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should be selected', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Chip title="My Button!" selected />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
