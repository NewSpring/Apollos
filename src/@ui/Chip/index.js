// Chippy!
import React from 'react';
import { compose, mapProps } from 'recompose';

// touchable native feedback currently is having flex layout issues
// on react-native android, so we fall back to TouchableOpacity
import { TouchableOpacity } from 'react-native';

import { H6 } from '@ui/typography';
import Button from '@ui/Button';
import Icon from '@ui/Icon';
import { withTheme } from '@ui/theme';
import styled from '@ui/styled';

export { default as ChipList } from './List';

const enhance = compose(
  withTheme(),
  mapProps(({ theme, selected, ...otherProps }) => ({
    type: selected ? 'primary' : 'default',
    iconSize: theme.sizing.baseUnit,
    ...otherProps,
  })),
);

const TitleText = styled(({ withIcon = false }) => ({
  ...(withIcon ? { flexGrow: 1 } : {}),
  textAlign: 'center',
  alignItems: 'center',
  justifyContent: 'center',
  paddingHorizontal: 6,
}))(H6);

const StyledButton = styled(({ theme, withIcon = false }) => ({
  justifyContent: withIcon ? 'flex-end' : 'center',
  paddingHorizontal: theme.sizing.baseUnit / 4,
  paddingVertical: theme.sizing.baseUnit / 4,
  height: theme.sizing.baseUnit * 2,
  marginRight: theme.sizing.baseUnit / 2,
  marginBottom: theme.sizing.baseUnit / 2,
}), 'Chip')(Button);

const Chip = enhance(({
  icon,
  iconStyles = {},
  iconSize,
  selected,
  title,
  pill = false,
  ...buttonProps
}) => (
  <StyledButton withIcon={icon} TouchableComponent={TouchableOpacity} pill={pill} {...buttonProps}>
    {title ? <TitleText withIcon={icon}>{title}</TitleText> : null}
    {icon ? <Icon name={icon} style={iconStyles} size={iconSize} /> : null}
  </StyledButton>
));

export default Chip;
