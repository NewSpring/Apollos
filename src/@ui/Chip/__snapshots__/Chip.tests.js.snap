// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Chip component should be selected 1`] = `
<View
  bordered={false}
  disabled={false}
  pill={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#6bac43",
      "borderColor": "#6bac43",
      "borderRadius": 6,
      "borderWidth": 2,
      "elevation": 2,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginRight": 10,
      "opacity": 1,
      "overflow": "hidden",
      "paddingHorizontal": 5,
      "paddingVertical": 5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignItems": "center",
        "color": "#ffffff",
        "fontFamily": "Colfax-Bold",
        "fontSize": 15.75,
        "justifyContent": "center",
        "lineHeight": 17.75,
        "paddingBottom": 1.8,
        "paddingHorizontal": 6,
        "paddingTop": 4.5,
        "textAlign": "center",
      }
    }
    withIcon={undefined}
  >
    My Button!
  </Text>
</View>
`;

exports[`The Chip component should render an icon 1`] = `
<View
  bordered={false}
  disabled={false}
  pill={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#dddddd",
      "borderColor": "#dddddd",
      "borderRadius": 6,
      "borderWidth": 2,
      "elevation": 2,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "flex-end",
      "marginBottom": 10,
      "marginRight": 10,
      "opacity": 1,
      "overflow": "hidden",
      "paddingHorizontal": 5,
      "paddingVertical": 5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignItems": "center",
        "color": "#303030",
        "flexGrow": 1,
        "fontFamily": "Colfax-Bold",
        "fontSize": 15.75,
        "justifyContent": "center",
        "lineHeight": 17.75,
        "paddingBottom": 1.8,
        "paddingHorizontal": 6,
        "paddingTop": 4.5,
        "textAlign": "center",
      }
    }
    withIcon="like"
  >
    Heart!!!
  </Text>
  <Svg
    height={20}
    style={Object {}}
    viewBox="0 0 24 24"
    width={20}
  >
    <Path
      d="M13.13 20.2h-2.26l1.4-1.4.03-.02.03-.03c6.37-5.6 8.06-7.67 8.06-10.27 0-2.2-1.7-3.9-3.9-3.9-1.23 0-2.48.6-3.28 1.54L12 7.57l-1.23-1.45C9.97 5.18 8.72 4.6 7.5 4.6c-2.2 0-3.9 1.67-3.9 3.88 0 2.6 1.7 4.66 8.07 10.27l.03.03.04.03 1.4 1.4zM10.8 4.03l-.03.03h.1c.4.3.8.64 1.13 1.03.33-.4.72-.75 1.14-1.05h.1c-.02 0-.03-.02-.04-.03.97-.66 2.13-1.03 3.3-1.03C19.6 3 22 5.4 22 8.47c0 3.8-3.4 6.88-8.6 11.46l-1.4 1.4-1.4-1.4C5.4 15.36 2 12.27 2 8.48 2 5.38 4.4 3 7.5 3c1.17 0 2.33.37 3.3 1.03z"
      fill="#303030"
    />
  </Svg>
</View>
`;

exports[`The Chip component should render correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    bordered={false}
    disabled={false}
    pill={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dddddd",
        "borderColor": "#dddddd",
        "borderRadius": 6,
        "borderWidth": 2,
        "elevation": 2,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginRight": 10,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 5,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "color": "#303030",
          "fontFamily": "Colfax-Bold",
          "fontSize": 15.75,
          "justifyContent": "center",
          "lineHeight": 17.75,
          "paddingBottom": 1.8,
          "paddingHorizontal": 6,
          "paddingTop": 4.5,
          "textAlign": "center",
        }
      }
      withIcon={undefined}
    >
      My Button!
    </Text>
  </View>
</View>
`;
